src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -64,55 +62,28 @@
  * about these suffixes are added.
  */
 public class SkroutzGreekStemmer {
-  public final static String DEFAULT_STOPWORD_FILE = "stopwords.txt";
-  private final CharArraySet stopwords;
-  private Map<String, char[]> stepZeroExceptions;
+  private CharArraySet protectedWords = CharArraySet.EMPTY_SET;
+  private CharArrayMap stepZeroExceptions = CharArrayMap.emptyMap();
 
-  public SkroutzGreekStemmer(final CharArraySet stopwords) {
-    this.stopwords = stopwords;
+  public SkroutzGreekStemmer() {
     try {
-      this.stepZeroExceptions = loadExceptions("step_0_exceptions");
+      this.protectedWords = new CharArraySet(
```

Comment:
[minor] Maybe `loadProtected` could return a `CharArraySet` (and equivalently `loadExceptions` a `CharArrayMap`), instead of handling this inside the constructor.